x/revenue/types/msg.go

```diff
@@ -171,3 +172,27 @@ func (msg MsgUpdateRevenue) GetSigners() []sdk.AccAddress {
 	from := sdk.MustAccAddressFromBech32(msg.DeployerAddress)
 	return []sdk.AccAddress{from}
 }
+
+// GetSigners returns the expected signers for a MsgUpdateParams message.
+func (m *MsgUpdateParams) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Authority)
+	return []sdk.AccAddress{addr}
+}
+
+// ValidateBasic does a sanity check of the provided data
+func (m *MsgUpdateParams) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Authority); err != nil {
+		return errortypes.Wrap(err, "invalid authority address")
```

Comment:
Refactor: use the new errors pkg
```suggestion
		return errorsmod.Wrap(err, "invalid authority address")
```